### PR TITLE
feature: added the new --main-conf option for specifying main

### PR DESCRIPTION
**Pull request by dndx**
configuration directives directly from command line.

### CODE REVIEW

File: bin/resty [hunk 1]
```diff
@@ -195,6 +200,16 @@ my $lua_shared_dicts = '';
                 next;
             }
 
+            if ($arg eq '--main-conf') {
+                my $v = shift @args;
+                if (!defined $v) {
+                    die "option --http-conf takes an argument but ",
```
Comment (agentzh) [inline bin/resty]: Copy&paste mistake here.

File: bin/resty [hunk 2]
```diff
@@ -185,16 +185,31 @@ my $lua_shared_dicts = '';
...
+            if ($arg =~ /^--main-conf=(.*)/) {
+                push @main_confs, $1;
+                next;
+            }
+
             if ($arg eq '--http-conf') {
                 my $v = shift @args;
                 if (!defined $v) {
-                    die "option --http-conf takes an argument but ",
+                    die "option --main-conf takes an argument but ",
```
Comment (doujiang24) [inline bin/resty]: unexpected change.

File: bin/resty [hunk 3]
```diff
@@ -912,6 +929,9 @@ Options:
 
     -l LIB              Require library "lib".
 
+    --main-conf CONF    Specifies nginx.conf snippet inserted into the nginx main
```
Comment (agentzh) [inline bin/resty]: Exceeding 80 cols.

Also need to update the usage copy in the README file.

**Comment by agentzh** [comment]
Merged with the following extra patch:

```diff
diff --git a/README.md b/README.md
index 4b83505..6e0f227 100644
--- a/README.md
+++ b/README.md
@@ -75,6 +75,10 @@ Options:

     -l LIB              Require library "lib".

+    --main-conf CONF    Specifies nginx.conf snippet inserted into the nginx
+                        main {} configuration block (multiple instances are
+                        supported).
+
     --main-include PATH Include the specified file in the nginx main
                         configuration block (multiple instances are supported).

diff --git a/bin/resty b/bin/resty
index 1e8c972..6eb81a4 100755
--- a/bin/resty
+++ b/bin/resty
@@ -929,8 +929,9 @@ Options:

     -l LIB              Require library "lib".

-    --main-conf CONF    Specifies nginx.conf snippet inserted into the nginx main
-                        configuration block (multiple instances are supported).
+    --main-conf CONF    Specifies nginx.conf snippet inserted into the nginx
+                        main {} configuration block (multiple instances are
+                        supported).

     --main-include PATH Include the specified file in the nginx main
                         configuration block (multiple instances are supported).
diff --git a/t/resty/options.t b/t/resty/options.t
index 4887216..4d0307e 100644
--- a/t/resty/options.t
+++ b/t/resty/options.t
@@ -72,8 +72,9 @@ Options:

     -l LIB              Require library "lib".

-    --main-conf CONF    Specifies nginx.conf snippet inserted into the nginx main
-                        configuration block (multiple instances are supported).
+    --main-conf CONF    Specifies nginx.conf snippet inserted into the nginx
+                        main {} configuration block (multiple instances are
+                        supported).

     --main-include PATH Include the specified file in the nginx main
                         configuration block (multiple instances are supported).
```

**Pull request closed without merging**